modules/pam_unix/support.c

```diff
@@ -242,23 +247,36 @@ int _set_ctrl(pam_handle_t *pamh, int flags, int *remember, int *rounds,
 		set(UNIX__NONULL, ctrl);
 	}
 
-	/* Set default rounds for blowfish */
```

Comment:
I do not really  understand why you need to do all these changes here. It would be better to just set rounds to 0 in pam_unix_passwd.c. There is apparently also a related bug in the existing code that when you get SHA_CRYPT_MAX_ROUNDS from login.defs it won't set the UNIX_ALGO_ROUNDS ctrl. But instead of setting it here it would be better to ignore it in the create_password_hash() completely and just check for round != 0 there.

Reply:
Fixed in rebased commit.